code/modules/vehicles/mecha/working/ripley.dm

```diff
@@ -232,19 +340,81 @@ GLOBAL_DATUM(cargo_ripley, /obj/vehicle/sealed/mecha/ripley/cargo)
 	if(.)
 		return TRUE
 	if(action == "eject")
-		var/obj/vehicle/sealed/mecha/ripley/miner = chassis
-		var/obj/crate = locate(params["cargoref"]) in miner.cargo
+		var/obj/crate = locate(params["cargoref"]) in contents
 		if(!crate)
 			return FALSE
-		to_chat(miner.occupants, "[icon2html(src,  miner.occupants)][span_notice("You unload [crate].")]")
+		to_chat(chassis.occupants, "[icon2html(src,  chassis.occupants)][span_notice("You unload [crate].")]")
 		crate.forceMove(drop_location())
-		LAZYREMOVE(miner.cargo, crate)
-		if(crate == miner.ore_box)
-			miner.ore_box = null
+		if(crate == chassis.ore_box)
+			chassis.ore_box = null
 		playsound(chassis, 'sound/weapons/tap.ogg', 50, TRUE)
-		log_message("Unloaded [crate]. Cargo compartment capacity: [miner.cargo_capacity - LAZYLEN(miner.cargo)]", LOG_MECHA)
+		log_message("Unloaded [crate]. Cargo compartment capacity: [cargo_capacity - contents.len]", LOG_MECHA)
 		return TRUE
 
+/obj/item/mecha_parts/mecha_equipment/ejector/seccage
+	name = "holding cell"
+	desc = "Holds suspects loaded with a hydraulic claw."
+	cargo_capacity = 4
+
+/obj/item/mecha_parts/mecha_equipment/ejector/seccage/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_MOB_REMOVING_CUFFS, PROC_REF(stop_cuff_removal))
```

Comment:
`src`? Is the sec cage going to be trying to remove its cuffs?
It seems like you'd want to register this on something entering the cage instead, no?

Reply:
I added a signal sent to a mob's loc when the mob tries to remove its cuffs. The equipment can just listen to signals sent to itself rather than having to juggle registering and unregistering mobs as they enter and leave

Reply:
If the signal is sent to the mob's loc it should not be called `COMSIG_MOB_REMOVING_CUFFS`, but also it makes significantly more sense for it to be sent by the mob rather than its location.